source/slang/slang-ir-autodiff-primal-hoist.cpp

```diff
@@ -500,6 +514,15 @@ RefPtr<HoistedPrimalsInfo> AutodiffCheckpointPolicyBase::processFunc(
                             callVarWorkList.add(callUser);
                     }
                 }
+
+                // This is a bit of a hack.. ideally we need to add the var to the worklist for
+                // further processing rather than replicating those operations here.
+                //
```

Comment:
Why can’t we just do as the comment said?

Reply:
I believe the body of the loop needs a small rewrite. I can do this in a follow-up PR.

Reply:
Opened an issue to track this cleanup: https://github.com/shader-slang/slang/issues/6214